versioned_docs/version-4.4/building-extensions/modules/module-development-tutorial/step7-javascript.md

```diff
@@ -0,0 +1,200 @@
+---
+sidebar_position: 7
+title: Step 7 Adding Javascript
+---
+
+Step 7 Adding Javascript
+========================
+
+## Introduction
+
+In this step we add a "!" to the end of the "Hello Username" greeting, but implement this using javascript, and pass the "!" as a variable from the PHP code. 
+So this step describes:
+- how to pass variables from PHP to javascript.
+- how to use the Joomla Web Asset Manager to manage the javascript asset
+
+The source code is available at [mod_hello step 7](https://github.com/joomla/manual-examples/tree/main/module-tutorial/step7_javascript). 
+
+## Passing Variables to Javascript
+
+To pass variables from PHP to js we use the [Document::addScriptOptions](https://api.joomla.org/cms-4/classes/Joomla-CMS-Document-Document.html#method_addScriptOptions) function. 
+
+```php
+$document = $this->app->getDocument();
+$document->addScriptOptions('vars', array('suffix' => "!"));

```

Comment:
I suggest to use `mod_hello.vars` as the options name (prefixed by extension name) for the tutorial.
It is not strict but recommended approach. Because it is global storage of the options and "very generic" option name can collide with another extension.

When people will copy/paste this tutorial, they will doing it "right" :wink:

Reply:
Thanks for the tip @Fedik, I'll put that in.